### DISCUSSION TITLE
Bilansowanie międzyfazowe

### DISCUSSION THREAD

**Discussion by oleksymaciek** [PiotrMachowski/Home-Assistant-custom-components-Tauron-AMIplus#188] [General]
Czy jest możliwość pobrania danych po zbilansowaniu międzyfazowym? W aplikacji Elicznik jest to dostępne po kliknięciu "energia po zbilansowaniu". Dane te byłyby idealne do obliczenia dokładnej ilości kWh pozostałych do wykorzystania (netmetering)

**Comment by PiotrMachowski** [comment]
Już są encje "zbalansowane", jest jakiś problem z nimi?

**Reply by oleksymaciek** [reply]
Chodzi o to że encje pobierane z aplikacji nie uwzględniają bilansowania międzyfazowego które tauron jakiś czas temu umieścił w aplikacji. Chodzi tu oczywiście o prosumentów. Czyli np. produkuję energię na jednej fazie a pobieram z drugiej. Wówczas encja energii  oddanej do sieci jest większa i pobranej
![Bilansowanie](https://github.com/user-attachments/assets/559d841d-1eb5-4a71-a2cf-2290470edd1d)
 również jest większa. Po zbilansowaniu energia i pobrana i oddana jest mniejsza i wtedy faktycznie można dokładnie obliczyć ile energii zostaje w "magazynie" (Energia oddana* 0,8/0,7 - energia pobrana). Na załączonym screenie to widać. Jest to screen z tego samego dnia.

**Reply by PiotrMachowski** [reply]
Czyli to nie jest to samo, co bilansowanie godzinowe? Nie znam się zbytnio na tym temacie, bo nie jestem prosumentem

**Comment by oleksymaciek** [comment]
Nie wiem jak jest rozwiązane bilansowanie godzinowe ale to jest raczej to samo

**Reply by PiotrMachowski** [reply]
Jeśli to jest to samo, to encje "zbalansowane" już zawierają te dane

**Comment by oleksymaciek** [comment]
Jeśli chodzi o encje z dopiskiem "balance" to jest to co innego niestety. Encje balance to różnica między energią wprowadzoną a pobraną. Dołączony skrypt który oblicza energię pozostałą w banku Taurona jest już dokładniejszy ale niestety nadal nie oblicza dokładnie. Trzeba by pobrać z aplikacji nowe encje "Energia po zbilansowaniu" i wtedy mozna dokładnie obliczyć energię pozostałą w banku

**Reply by PiotrMachowski** [reply]
Proszę odpisuj w tym samym wątku.

> Encje balance to różnica między energią wprowadzoną a pobraną.

Nie do końca, takie różnice są obliczane osobno dla każdej godziny, a stan encji to suma tych różnic. W atrybutach dostępne są wartości składowe dla konsumpcji i generacji.
